src/common/io/socket/session.c

```diff
@@ -86,36 +82,37 @@ sckSessionNew(SocketSessionType type, int fd, const String *host, unsigned int p
 
 /**********************************************************************************************************************************/
 void
-sckSessionReadWait(SocketSession *this)
+sckSessionReadyRead(SocketSession *this)
 {
     FUNCTION_LOG_BEGIN(logLevelTrace);
         FUNCTION_LOG_PARAM(SOCKET_SESSION, this);
     FUNCTION_LOG_END();
 
     ASSERT(this != NULL);
-    ASSERT(this->fd != -1);
 
-    // Initialize the file descriptor set used for select
-    fd_set selectSet;
-    FD_ZERO(&selectSet);
+    if (!sckReady(this->fd, true, false, this->timeout))
```

Comment:
I don't understand why we have sckReadyRead when we don't use it anywhere - especially here in sckSessionReadyRead.  Ditto for sckReadyWrite (although it is used one time in the code but not in sckSessionReadyWrite).

Reply:
Good point. sckReadyRead() and sckReadWrite() should be used in SocketSession. This is a copy-pasto from an earlier version of the code. sckReadWrite() is also used independently in the TLS non-block code, but sckReadyRead() is not. Figured it made sense to keep it symmetric, though.

socket/session.c udpate, let me know when I should push that.